code/__HELPERS/unsorted.dm

```diff
@@ -162,6 +162,7 @@ Turf and target are separate in case you want to teleport some distance from a t
 				py+=sdy
 			px+=sdx		//Step on in x direction
 			line+=locate(px,py,M.z)//Add the turf to the list
+			CHECK_TICK
```

Comment:
Nothing better than random sleep in my helper function.

Reply:
It's not a critical function, it can handle being sidelined for a few seconds.

Reply:
Something critical might use it though, making this ALSO critical. I have serious doubts you made sure that doesn't happen.
You also need to check if something got deleted after you call this because you can't be certain that nothing got deleted afterwards.
You can't just slap a sleep in a random helper function that might get used all over the place and not expect bad shit to happen. Instead slap the check tick into the actual mob code.

Also just use block instead of spiralturfs, it's incredibly cheap and also consistent with previous behaviour.

Reply:
Hell, do you even know what check tick does? It does not sideline a proc for a few seconds, it makes it skip exactly one tick. Since this particular proc will never take more than a fraction of a tick, this will just extremely rarely randomly sleep in the middle of it.

Hell, there is no reason why your sleep would actually be any cheaper than the inbuilt byond version. It isn't even optimized in the slightest and calls can_see on the same object multiple times although there would be trivial ways to prevent that.

The FindTarget stuff seems fine enough but this is just braindead.